smoke-test/tests/privileges/test_privileges.py

```diff
@@ -450,4 +450,64 @@ def test_privilege_to_create_and_manage_policies():
 
 
     # Ensure that user can't create a policy after privilege is removed by admin
-    _ensure_cant_perform_action(user_session, create_policy,"createPolicy")
\ No newline at end of file
+    _ensure_cant_perform_action(user_session, create_policy,"createPolicy")
+
+
+@pytest.mark.dependency(depends=["test_healthchecks"])
+def test_privilege_from_group_role_can_create_and_manage_tokens():
+
+    (admin_user, admin_pass) = get_admin_credentials()
+    admin_session = login_as(admin_user, admin_pass)
+    user_session = login_as("user", "user")
+    secret_urn = "urn:li:dataHubSecret:TestSecretName"
+
+    # Verify new user can't create secrets
+    create_secret = {
+        "query": """mutation createSecret($input: CreateSecretInput!) {\n
+            createSecret(input: $input)\n}""",
+        "variables": {
+            "input":{
+                "name":"TestSecretName",
+                "value":"Test Secret Value",
+                "description":"Test Secret Description"
+            }
+        },
+    }
+    _ensure_cant_perform_action(user_session, create_secret,"createSecret")
+
+    # Create group and grant it the admin role.
+    group_urn = create_group(admin_session, "Test Group")
+
+    # Assign admin role to group
+    assign_role(admin_session,"urn:li:dataHubRole:Admin", [group_urn])
+
+    # Assign user to group
+    assign_user_to_group(admin_session, group_urn, ["urn:li:corpuser:user"])
+
+    # Verify new user with admin group can create and manage secrets
+    # Create a secret
+    _ensure_can_create_secret(user_session, create_secret, secret_urn)
```

Comment:
This is where the previous operation passes now where before it fails.
This test code was derived from a different unit test in the same file `test_privilege_to_create_and_manage_secrets`. The reason I used the same test is that managing secrets is an admin role only operation. If the user is an admin, he/she (or the group) can perform it, otherwise no.